File: jobportal/src/main/java/com/kiwisoft/jobportal/service/UserService.java
package com.kiwisoft.jobportal.service;

import com.kiwisoft.jobportal.entity.User;
import com.kiwisoft.jobportal.dto.response.UserProfileResponse;
import com.kiwisoft.jobportal.dto.request.UserRequest;
import com.kiwisoft.jobportal.dto.response.UserResponse;
import com.kiwisoft.jobportal.dto.request.UpdateUserRequest;
import java.util.List;

public interface UserService {
    UserProfileResponse getProfile(String email);

    List<User> getAllUsers();


    User getUserById(Long id);
    UserResponse createUser(UserRequest request);

    UserResponse updateUser(Long id, UpdateUserRequest request);


    void deleteUser(Long id);
    void activateUser(Long id);
}

File: jobportal-frontend/src/services/userService.js
import api from "./api";

export const getAllUsers = async () => {
  const res = await api.get("/users");
  return res.data;
};

export const getUserById = async (id) => {
  const res = await api.get(`/users/${id}`);
  return res.data;
};

export const createUser = async (data) => {
  const res = await api.post("/users", data);
  return res.data;
};

export const updateUser = async (id, data) => {
  const res = await api.put(`/users/${id}`, data);
  return res.data;
};

export const deleteUser = async (id) => {
  const res = await api.delete(`/users/${id}`);
  return res.data;
};

export const activateUser = async (id) => {
  const res = await api.patch(`/users/${id}/activate`);
  return res.data;
};